smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -128,6 +132,8 @@ contract MixinTransfer is
     if (toKey.tokenId == 0) {
       toKey.tokenId = fromKey.tokenId;
       _recordOwner(_recipient, toKey.tokenId);
+      keyManagerOf[toKey.tokenId] = _recipient;
```

Comment:
Again what about the expired scenario?

Reply:
the check for this block is:
`if (toKey.tokenId == 0) {...`
ensuring this only operates on new keys.